charts/streams-app/templates/deployment.yaml

```diff
@@ -60,50 +60,23 @@ spec:
       serviceAccountName: {{ .Values.serviceAccountName }}
       {{- end }}
       {{- if .Values.tolerations }}
-      tolerations:
-{{ toYaml .Values.tolerations | indent 8 }}
+      tolerations: {{ toYaml .Values.tolerations | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity: {{- toYaml . | nindent 8 }}
       {{- end }}
-      affinity:
-        {{- if .Values.nodeAffinity }}
-        nodeAffinity:
-{{ toYaml .Values.nodeAffinity | indent 10 }}
-        {{- end }}
-        podAntiAffinity:
-        {{- if .Values.requirePodAntiAffinity }}
-          requiredDuringSchedulingIgnoredDuringExecution:
-          - labelSelector:
-            matchExpressions:
-            - key: app
-              operator: In
-              values:
-              - {{ template "streams-app.name" . }}
```

Comment:
Since this is a dynamic value injected by helm, I don't see how we can replicate that with the new way

Reply:
You can achieve the same if we use `tpl` in the deployment.yaml, e.g.:
```yaml
      {{- if .Values.affinity }}
      affinity: 
        {{- tpl (toYaml .Values.affinity) . | nindent 8 }}
      {{- end }}
```

and the values.yaml:
```yaml
  podAntiAffinity:
     preferredDuringSchedulingIgnoredDuringExecution:
      - weight: 1
        podAffinityTerm:
          topologyKey: "kubernetes.io/hostname"
          labelSelector:
            matchExpressions:
              - key: app
                operator: In
                values:
                  - '{{ .Values.nameOverride }}'
```